modes/snake/evil-collection-snake.el

```diff
@@ -0,0 +1,54 @@
+;;; evil-collection-snake.el --- Bindings for `snake' -*- lexical-binding: t -*-
+
+;; Copyright (C) 2021 Farzin Firouzi
+
+;; Author: Farzin Firouzi
+;; Maintainer: Farzin Firouzi
+;; Farzin Firouzi <farzinf@riseup.net>
+;; URL: https://github.com/emacs-evil/evil-collection
+;; Version: 0.0.1
+;; Package-Requires: ((emacs "25.1"))
+;; Keywords: evil, emacs, tools
+
+;; This program is free software; you can redistribute it and/or modify
+;; it under the terms of the GNU General Public License as published by
+;; the Free Software Foundation, either version 3 of the License, or
+;; (at your option) any later version.
+
+;; This program is distributed in the hope that it will be useful,
+;; but WITHOUT ANY WARRANTY; without even the implied warranty of
+;; MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+;; GNU General Public License for more details.
+
+;; You should have received a copy of the GNU General Public License
+;; along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+;;; Commentary:
+;; Bindings for `snake'.
+
+;;; Code:
+(require 'evil-collection)
+(require 'snake nil t)
```

Comment:
`snake` is a builtin package, the `FILENAME` and `NOERROR` parameters can be omitted..

Reply:
@condy0919 copied the tetris dir and modified the appropriate files. Is there a reason why `(require 'tetris nil t)` is included in the tetris dir?

Reply:
Also, can you please provide the steps, or provide a resource I can use to test this code? This is my first pull request. I know the steps for evil-mode is to `cd` into the dir and run `make emacs`. How do I do this for evil-collection? I noticed a make file. Do I use that?

Reply:
> Also, can you please provide the steps, or provide a resource I can use to test this code? This is my first pull request. I know the steps for evil-mode is to `cd` into the dir and run `make emacs`. How do I do this for evil-collection? I noticed a make file. Do I use that?

Manually <kbd>M-:</kbd> `(evil-collection-snake-setup)`

Reply:
> @condy0919 copied the tetris dir and modified the appropriate files. Is there a reason why `(require 'tetris nil t)` is included in the tetris dir? here: https://github.com/emacs-evil/evil-collection/blob/master/modes/tetris/evil-collection-tetris.el

It should also be fixed.

Reply:
In checking other builtin packages other than `tetris` (example: dired, eww, and virtually all of them) it appears they too have a line equivalent to `line 30 and 31` above, despite it being a builtin package. I understand you're point, and while I am happy to go forward with changing all files in `/modes/` that are related to built-ins, it would be a good idea to check with others to see if this is a good idea. Does `(require '...)` even take that much time to eval/parse?

Reply:
> while I am happy to go forward with changing all files in /modes/ that are related to built-ins

It seems like only `tetris` is the one should be fixed.

> Does (require '...) even take that much time to eval/parse?

It depends. If you found all commands used in `evil-collection-xxx.el` are `autoload`ed, then the `(require '...)` should be removed for better user experience.

e.g. https://github.com/emacs-evil/evil-collection/blob/master/modes/consult/evil-collection-consult.el#L37